api/admin/controller/report.py

```diff
@@ -0,0 +1,42 @@
+import json
+import logging
+from dataclasses import asdict
+from http import HTTPStatus
+
+import flask
+from flask import Response
+
+from api.controller.circulation_manager import CirculationManagerController
+from core.model import Library
+from core.model.admin import Admin
+from core.model.asynctask import AsyncTaskType, InventoryReportTaskData, queue_task
+from core.util.problem_detail import ProblemDetail, ProblemDetailException
+
+
+class ReportController(CirculationManagerController):
```

Comment:
The test in test_report seems to rely on the CirculationManagerController.  I left it in there.

Reply:
```diff
diff --git a/api/admin/controller/__init__.py b/api/admin/controller/__init__.py
index ddf6fef3a..83eaaf4cb 100644
--- a/api/admin/controller/__init__.py
+++ b/api/admin/controller/__init__.py
@@ -64,4 +64,4 @@ def setup_admin_controllers(manager: CirculationManager):
     manager.admin_announcement_service = AnnouncementSettings(manager._db)
     manager.admin_search_controller = AdminSearchController(manager)
     manager.admin_quicksight_controller = QuickSightController(manager)
-    manager.admin_report_controller = ReportController(manager)
+    manager.admin_report_controller = ReportController(manager._db)
diff --git a/api/admin/controller/report.py b/api/admin/controller/report.py
index ccb182681..97c9463fa 100644
--- a/api/admin/controller/report.py
+++ b/api/admin/controller/report.py
@@ -4,6 +4,7 @@ from http import HTTPStatus

 import flask
 from flask import Response
+from sqlalchemy.orm import Session

 from api.controller.circulation_manager import CirculationManagerController
 from core.model import Library
@@ -14,10 +15,14 @@ from core.model.deferredtask import (
     queue_task,
 )
 from core.problem_details import INTERNAL_SERVER_ERROR
+from core.util.log import LoggerMixin
 from core.util.problem_detail import ProblemDetail


-class ReportController(CirculationManagerController):
+class ReportController(LoggerMixin):
+    def __init__(self, db: Session):
+        self._db = db
+
     def generate_inventory_report(self) -> Response | ProblemDetail:
         library: Library = getattr(flask.request, "library")
         admin: Admin = getattr(flask.request, "admin")
```

I think that patch removes the requirement on `CirculationManagerController`. There is a lot of code in `CirculationManagerController` and its parent class `BaseCirculationManagerController` that would be nice to avoid pulling into this controller if we don't need.